modules/sdk-coin-polygon/test/unit/polygon.ts

```diff
@@ -763,4 +763,164 @@ describe('Polygon', function () {
     });
   });
+
+  describe('Evm Based Cross Chain Recovery transaction:', function () {
+    const baseUrl = 'https://api-testnet.polygonscan.com';
+    const userXpub =
+      'xpub661MyMwAqRbcEeTc8789MK5PUGEYiPG4F4V17n2Rd2LoTATA1XoCnJT5FAYAShQxSxtFjpo5NHmcWwTp2LiWGBMwpUcAA3HywhxivgYfq7q';
+
+    it('should generate an unsigned recovery txn for cold wallet', async function () {
+      const walletContractAddress = TestBitGo.V2.TEST_ETH_WALLET_FIRST_ADDRESS as string;
+      const feeAddress = '0x33a42faea3c6e87021347e51700b48aaf49aa1e7';
+      const destinationAddress = '0xd5ADdE17feD8baed3F32b84AF05B8F2816f7b560';
+      const bitgoFeeAddress = '0xE5986CE4490Deb67d2950562Ceb930Ddf9be7a14';
+
+      const basecoin = bitgo.coin('tpolygon') as Polygon;
+      nock(baseUrl).get('/api').query(mockData.getTxListRequest(feeAddress)).reply(200, mockData.getTxListResponse);
+      nock(baseUrl).get('/api').query(mockData.getBalanceRequest(feeAddress)).reply(200, mockData.getBalanceResponse);
+      nock(baseUrl)
+        .get('/api')
+        .query(mockData.getBalanceRequest(walletContractAddress))
+        .reply(200, mockData.getBalanceResponse);
+      nock(baseUrl).get('/api').query(mockData.getContractCallRequest).reply(200, mockData.getContractCallResponse);
+
+      const transaction = (await basecoin.recover({
+        userKey: userXpub,
+        backupKey: '',
+        walletContractAddress: walletContractAddress,
+        feeAddress: feeAddress,
+        recoveryDestination: destinationAddress,
+        eip1559: { maxFeePerGas: 20000000000, maxPriorityFeePerGas: 10000000000 },
+        gasLimit: 500000,
+        bitgoFeeAddress: bitgoFeeAddress,
+      })) as OfflineVaultTxInfo;
+
+      should.exist(transaction);
+      transaction.should.have.property('txHex');
+      transaction.should.have.property('userKey');
+      transaction.should.have.property('coin');
+      transaction.should.have.property('contractSequenceId');
+      transaction.should.have.property('expireTime');
+      transaction.should.have.property('gasLimit');
+      transaction.gasLimit.should.equal('500000');
+      transaction.should.have.property('isEvmBasedCrossChainRecovery');
+      transaction.isEvmBasedCrossChainRecovery?.should.equal(true);
+      transaction.should.have.property('walletContractAddress');
+      transaction.walletContractAddress.should.equal(TestBitGo.V2.TEST_ETH_WALLET_FIRST_ADDRESS);
+      transaction.should.have.property('recipients');
+      const recipient = transaction.recipients[0];
+      recipient.should.have.property('address');
+      recipient.address.should.equal(destinationAddress);
+      recipient.should.have.property('amount');
+      recipient.amount.should.equal('9999999999999999928');
+      transaction.should.have.property('amount');
+      transaction.amount.should.equal('9999999999999999928');
+    });
+
+    it('should generate an unsigned recovery txn for custody wallet', async function () {
```

Comment:
What params are different for cold and custody wallet?

Reply:
user public key will be provided in cold wallet but not in custody wallet.